### PR TITLE
Connection management - Zendesk

### PR DESCRIPTION
**Pull request by Shubhangi-cs**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -290,6 +290,11 @@
       <version>${zendesk-java-client.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>io.cdap.cdap</groupId>
+      <artifactId>cdap-etl-api</artifactId>
+      <version>6.6.0-SNAPSHOT</version>
+    </dependency>
```
Comment (sgarg-CS) [inline pom.xml]: Why are we using two different versions for the same artifact `cdap-etl-api` ?

Comment (sgarg-CS) [reply pom.xml]: Also, we shouldn't be using the SNAPSHOT version.

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
```
Comment (sgarg-CS) [inline src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java]: Change to 2022

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
```
Comment (sgarg-CS) [inline src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java]: 2022

File: pom.xml (hunk 1 above)
Comment (Shubhangi-cs) [reply pom.xml]: Done the change.

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java (hunk 2 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java]: Done

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java (hunk 3 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java]: Done

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      HttpClientContext httpClientContext = HttpUtil.createHttpContext(this, countURL);
+      HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+      CloseableHttpClient httpClient = httpClientBuilder.build();
+      CloseableHttpResponse response = null;
+      try {
+        response = httpClient.execute(new HttpGet(countURL), httpClientContext);
+      } catch (IOException e) {
+        LOG.error("Unable to generate response", e);
+      }
+      StatusLine statusLine = response.getStatusLine();
```
Comment (vikasrathee-cs) [inline src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java:148]: if code will throw IOException, there will null response and will get NullPointerException here. Add error message in FailureCollector inside catch block and return from there.

File: src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java [hunk 5]
```diff
@@ -235,22 +236,23 @@ public void validate(FailureCollector collector) {
 
   @VisibleForTesting
   void validateConnection(FailureCollector collector) {
```
Comment (vikasrathee-cs) [inline src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java]: move this method to connector config class

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java (hunk 4 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java:148]: Done

File: src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java (hunk 5 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java]: This method is different than the method used for validation in ConnectorConfig. This needs to be placed here only.

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    try {
+      config.validateConnectionParameters(collector);
+    } catch (IOException e) {
+      LOG.error("Unable to validate connection", e);
+    }
+
+  }
```
Comment (Vipinofficial11) [inline src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java:113]: please remove the extra space.

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      int statusCode = statusLine.getStatusCode();
+      if (statusCode == HttpURLConnection.HTTP_NOT_FOUND) {
+        collector.addFailure("Subdomain is incorrect.", "Please check the value of subdomain");
+      } else if (statusCode == HttpURLConnection.HTTP_UNAUTHORIZED) {
+        collector.addFailure("Admin Email or Api endpoint is incorrect.", "Please check the values");
+      }
+    });
+  }
+
+}
```
Comment (Vipinofficial11) [inline src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java:161]: extra space.

File: src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java (hunk 5 above)
Comment (vikasrathee-cs) [reply src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java]: All 6 properties checked here should be part of connector config and this method should be part of that class.

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java (hunk 7 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java:161]: removed

File: src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java (hunk 6 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnector.java:113]: removed

File: src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java (hunk 5 above)
Comment (Shubhangi-cs) [reply src/main/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfig.java]: Done

**Pull request closed without merging**